ios/Classes/SwiftAirshipPlugin.swift

```diff
@@ -18,37 +18,53 @@ public class SwiftAirshipPlugin: NSObject, FlutterPlugin {
         .notificationStatusChanged: "com.airship.flutter/event/notification_status_changed"
     ]
 
-    private static let streams: [AirshipProxyEventType: AirshipEventStream] = {
+
+    private static let eventSubject = PassthroughSubject<AirshipProxyEventType, Never>()
+    private static let eventTask: Task<Void, Never> =  Task {
+        let pending = await AirshipProxyEventEmitter.shared.pendingEventAdded
+        for await event in pending {
+            eventSubject.send(event.type)
```

Comment:
We should be able to remove this now that `AirshipProxyEventEmitter` exposes a subject for pending

Reply:
You need to remove the publisher as well here and then have the streams use AIrshipProxyEventEmitter.shared.eventSubject (i should of exposed a publisher but oh well, its internal).